src/app/visualization/three-fiber-components/CanvasContents.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { useState } from 'react';
+import { useFrame } from 'react-three-fiber';
+import { Vector3 } from 'three';
+import {
+  cameraOrbitRadius,
+  cameraOrbitHeight,
+  upDirection,
+  GameState,
+  GameCoords,
+  Encampment,
+  ResourceNode,
+} from 'src/game';
+// TODO: better names for types / components / filenames / imports
+import EncampmentComponent from './Encampment';
+import GroundComponent from './Ground';
+import ResourceNodeComponent from './ResourceNode';
+
+type CanvasProps = { gameState: GameState };
+type Milliseconds = number;
+
+const offsetToCameraCoords = (offset: Milliseconds): number[] => {
+  const radians = offset;
+  const x = Math.cos(radians) * cameraOrbitRadius;
+  const y = Math.sin(radians) * cameraOrbitRadius;
+  return [x, y, cameraOrbitHeight]; // may need to fiddle
+};
+
+const renderArgsFromGameState = (
+  gameState: GameState,
+): { encampmentArgs: Encampment; resourceNodeArgsList: ResourceNode[] } => {
+  // remove placeholder once locations are implemented in main game state
```

Comment:
Let's discuss tomorrow but I think it'd be nice to make issues of these and the other TODOs so that we don't lose them over time.

Reply:
Great suggestion, I'll create an issue for this we can iterate on.